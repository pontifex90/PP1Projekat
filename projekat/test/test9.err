ERROR 2017-02-11 06:19:51,985 [main][MJParser.java:778] : Semantic error on line 6: non-void method must return a type.
ERROR 2017-02-11 06:19:51,986 [main][MJParser.java:778] : Semantic error on line 10: void methods shouldn't return a type.
ERROR 2017-02-11 06:19:51,988 [main][MJParser.java:778] : Semantic error on line 18: cannot assign to constant, method, class name or program name.
Greska: Na levoj strani dodele mora biti promenljiva!
ERROR 2017-02-11 06:19:51,989 [main][MJParser.java:778] : Semantic error on line 19: invalid number of parameters for method "someMethod2".
ERROR 2017-02-11 06:19:51,990 [main][MJParser.java:778] : Semantic error on line 20: invalid number of parameters for method "someMethod".
ERROR 2017-02-11 06:19:51,991 [main][MJParser.java:778] : Semantic error on line 21: incompatible types.
ERROR 2017-02-11 06:19:51,991 [main][MJParser.java:778] : Semantic error on line 21: formal ("b") and actual parameter types mismatch.
ERROR 2017-02-11 06:19:51,992 [main][MJParser.java:778] : Semantic error on line 22: incompatible types.
ERROR 2017-02-11 06:19:51,992 [main][MJParser.java:778] : Semantic error on line 23: wrong indexing expression type. It must be "int"
ERROR 2017-02-11 06:19:51,993 [main][MJParser.java:778] : Semantic error on line 24: operation can be done only with integers.
ERROR 2017-02-11 06:19:51,994 [main][MJParser.java:778] : Semantic error on line 26: incompatible types.
ERROR 2017-02-11 06:19:51,997 [main][MJParser.java:778] : Semantic error on line 29: expression in condition must be of boolean type.
ERROR 2017-02-11 06:19:51,997 [main][MJParser.java:778] : Semantic error on line 30: break/continue can be used only inside the for loop.
ERROR 2017-02-11 06:19:51,997 [main][MJParser.java:778] : Semantic error on line 31: break/continue can be used only inside the for loop.
ERROR 2017-02-11 06:19:51,998 [main][MJParser.java:778] : Semantic error on line 33: booleans, classes and arrays can only use "==" and "!=" relative operators.
ERROR 2017-02-11 06:19:51,998 [main][MJParser.java:778] : Semantic error on line 33: tried to compare incompatible types.
ERROR 2017-02-11 06:19:51,999 [main][MJParser.java:778] : Semantic error on line 36: booleans, classes and arrays can only use "==" and "!=" relative operators.
ERROR 2017-02-11 06:19:51,999 [main][MJParser.java:778] : Semantic error on line 36: tried to compare incompatible types.
ERROR 2017-02-11 06:19:51,999 [main][MJParser.java:778] : Semantic error on line 36: booleans, classes and arrays can only use "==" and "!=" relative operators.
ERROR 2017-02-11 06:19:51,999 [main][MJParser.java:778] : Semantic error on line 36: tried to compare incompatible types.
ERROR 2017-02-11 06:19:52,000 [main][MJParser.java:625] : All error messages follow:
				Semantic error on line 6: non-void method must return a type.
				Semantic error on line 10: void methods shouldn't return a type.
				Semantic error on line 18: cannot assign to constant, method, class name or program name.
				Semantic error on line 19: invalid number of parameters for method "someMethod2".
				Semantic error on line 20: invalid number of parameters for method "someMethod".
				Semantic error on line 21: incompatible types.
				Semantic error on line 21: formal ("b") and actual parameter types mismatch.
				Semantic error on line 22: incompatible types.
				Semantic error on line 23: wrong indexing expression type. It must be "int"
				Semantic error on line 24: operation can be done only with integers.
				Semantic error on line 26: incompatible types.
				Semantic error on line 29: expression in condition must be of boolean type.
				Semantic error on line 30: break/continue can be used only inside the for loop.
				Semantic error on line 31: break/continue can be used only inside the for loop.
				Semantic error on line 33: booleans, classes and arrays can only use "==" and "!=" relative operators.
				Semantic error on line 33: tried to compare incompatible types.
				Semantic error on line 36: booleans, classes and arrays can only use "==" and "!=" relative operators.
				Semantic error on line 36: tried to compare incompatible types.
				Semantic error on line 36: booleans, classes and arrays can only use "==" and "!=" relative operators.
				Semantic error on line 36: tried to compare incompatible types.
ERROR 2017-02-11 06:19:52,000 [main][MJParser.java:626] : Parsing finished with ERRORS!
