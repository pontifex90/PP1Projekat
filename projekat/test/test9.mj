
program test9

{	
	int someMethod(int a, int b) {
		return;
	}
	
	void someMethod2() {
		return 3;
	}

	void main()
	int a[];
	int b;
	bool f;
	{	
		someMethod2 = true;
		someMethod2(a, b);
		someMethod(1);
		someMethod(1, true);
		a = someMethod(2,3);
		a[a] = 3;
		a++;
		a[2 + 8] = 18;
		a[2 + 8] = true;
		a = a;
		a[a[a[a[2]]]] = 10;
		if(2 + 2) {
			break;
			continue;
		}
		if(f > 3) {
		
		}
		for(;a > 0 && a < 3;) {
			break;
		}
	}
}

