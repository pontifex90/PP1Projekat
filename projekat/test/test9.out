INFO  2017-02-11 06:19:51,966 [main][MJParser.java:615] : Compiling source file: D:\PP1ProjekatRepo\PP1Projekat\projekat\test\test9.mj
INFO  2017-02-11 06:19:51,978 [main][MJParser.java:588] : #2 program
INFO  2017-02-11 06:19:51,978 [main][MJParser.java:588] : #49 test9
INFO  2017-02-11 06:19:51,978 [main][MJParser.java:588] : #7 {
DEBUG 2017-02-11 06:19:51,980 [main][MJParser.java:770] : Semantic info on line 2: added program "test9" to symbol table. [ kind: 6  name: test9  type: 0]
INFO  2017-02-11 06:19:51,982 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:51,982 [main][MJParser.java:588] : #49 someMethod
INFO  2017-02-11 06:19:51,982 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,982 [main][MJParser.java:770] : Semantic info on line 5: added  method "someMethod" to symbol table. [ kind: 3  name: someMethod  type: 1]
INFO  2017-02-11 06:19:51,982 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:51,982 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,983 [main][MJParser.java:588] : #12 ,
DEBUG 2017-02-11 06:19:51,983 [main][MJParser.java:770] : Semantic info on line 5: added formal parameter variable "a" to symbol table. [ kind: 1  name: a  type: 1]
INFO  2017-02-11 06:19:51,983 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:51,983 [main][MJParser.java:588] : #49 b
INFO  2017-02-11 06:19:51,983 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:19:51,984 [main][MJParser.java:770] : Semantic info on line 5: added formal parameter variable "b" to symbol table. [ kind: 1  name: b  type: 1]
INFO  2017-02-11 06:19:51,984 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:51,984 [main][MJParser.java:588] : #27 return
INFO  2017-02-11 06:19:51,985 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,985 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:51,985 [main][MJParser.java:588] : #17 void
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #49 someMethod2
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,986 [main][MJParser.java:770] : Semantic info on line 9: added  method "someMethod2" to symbol table. [ kind: 3  name: someMethod2  type: 0]
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #27 return
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #17 void
INFO  2017-02-11 06:19:51,986 [main][MJParser.java:588] : #49 main
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,987 [main][MJParser.java:770] : Semantic info on line 13: added  method "main" to symbol table. [ kind: 3  name: main  type: 0]
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #13 [
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:51,987 [main][MJParser.java:770] : Semantic info on line 14: added local array variable "a" to symbol table. [ kind: 1  name: a  type: 3]
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #49 int
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #49 b
INFO  2017-02-11 06:19:51,987 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:51,987 [main][MJParser.java:770] : Semantic info on line 15: added local variable "b" to symbol table. [ kind: 1  name: b  type: 1]
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #49 bool
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #49 f
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:51,988 [main][MJParser.java:770] : Semantic info on line 16: added local variable "f" to symbol table. [ kind: 1  name: f  type: 5]
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #49 someMethod2
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:51,988 [main][MJParser.java:770] : Usage info on line 18: global method "someMethod2" used. [ kind:3  name: someMethod2  type: 0]
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #51 true
INFO  2017-02-11 06:19:51,988 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #49 someMethod2
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,989 [main][MJParser.java:770] : Usage info on line 19: global method "someMethod2" used. [ kind:3  name: someMethod2  type: 0]
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #12 ,
DEBUG 2017-02-11 06:19:51,989 [main][MJParser.java:770] : Usage info on line 19: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #49 b
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #10 )
DEBUG 2017-02-11 06:19:51,989 [main][MJParser.java:770] : Usage info on line 19: local variable "b" used. [ kind:1  name: b  type: 1]
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,989 [main][MJParser.java:588] : #49 someMethod
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,990 [main][MJParser.java:770] : Usage info on line 20: global method "someMethod" used. [ kind:3  name: someMethod  type: 1]
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #48 1
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #49 someMethod
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,990 [main][MJParser.java:770] : Usage info on line 21: global method "someMethod" used. [ kind:3  name: someMethod  type: 1]
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #48 1
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #12 ,
INFO  2017-02-11 06:19:51,990 [main][MJParser.java:588] : #51 true
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:51,991 [main][MJParser.java:770] : Usage info on line 22: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #49 someMethod
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #9 (
DEBUG 2017-02-11 06:19:51,991 [main][MJParser.java:770] : Usage info on line 22: global method "someMethod" used. [ kind:3  name: someMethod  type: 1]
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #48 2
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #12 ,
INFO  2017-02-11 06:19:51,991 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,992 [main][MJParser.java:770] : Usage info on line 23: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #14 ]
DEBUG 2017-02-11 06:19:51,992 [main][MJParser.java:770] : Usage info on line 23: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #6 =
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:19:51,992 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #19 ++
DEBUG 2017-02-11 06:19:51,993 [main][MJParser.java:770] : Usage info on line 24: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,993 [main][MJParser.java:770] : Usage info on line 25: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #48 2
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #28 +
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #48 8
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:19:51,993 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:51,993 [main][MJParser.java:770] : Usage info on line 25: element of array "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #48 18
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,994 [main][MJParser.java:770] : Usage info on line 26: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #48 2
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #28 +
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #48 8
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:51,994 [main][MJParser.java:770] : Usage info on line 26: element of array "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #51 true
INFO  2017-02-11 06:19:51,994 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:51,995 [main][MJParser.java:770] : Usage info on line 27: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #11 ;
DEBUG 2017-02-11 06:19:51,995 [main][MJParser.java:770] : Usage info on line 27: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,995 [main][MJParser.java:770] : Usage info on line 28: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,995 [main][MJParser.java:770] : Usage info on line 28: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,995 [main][MJParser.java:770] : Usage info on line 28: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,995 [main][MJParser.java:588] : #13 [
DEBUG 2017-02-11 06:19:51,996 [main][MJParser.java:770] : Usage info on line 28: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #48 2
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #14 ]
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #14 ]
DEBUG 2017-02-11 06:19:51,996 [main][MJParser.java:770] : Usage info on line 28: element of array "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #14 ]
DEBUG 2017-02-11 06:19:51,996 [main][MJParser.java:770] : Usage info on line 28: element of array "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #14 ]
DEBUG 2017-02-11 06:19:51,996 [main][MJParser.java:770] : Usage info on line 28: element of array "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #6 =
DEBUG 2017-02-11 06:19:51,996 [main][MJParser.java:770] : Usage info on line 28: element of array "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #48 10
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #38 if
INFO  2017-02-11 06:19:51,996 [main][MJParser.java:588] : #9 (
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #48 2
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #28 +
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #48 2
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #26 break
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #25 continue
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #38 if
INFO  2017-02-11 06:19:51,997 [main][MJParser.java:588] : #9 (
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #49 f
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #44 >
DEBUG 2017-02-11 06:19:51,998 [main][MJParser.java:770] : Usage info on line 33: local variable "f" used. [ kind:1  name: f  type: 5]
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #22 for
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #9 (
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,998 [main][MJParser.java:588] : #44 >
DEBUG 2017-02-11 06:19:51,999 [main][MJParser.java:770] : Usage info on line 36: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #48 0
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #40 &&
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #49 a
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #46 <
DEBUG 2017-02-11 06:19:51,999 [main][MJParser.java:770] : Usage info on line 36: local variable "a" used. [ kind:1  name: a  type: 3]
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #48 3
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #10 )
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #7 {
INFO  2017-02-11 06:19:51,999 [main][MJParser.java:588] : #26 break
INFO  2017-02-11 06:19:52,000 [main][MJParser.java:588] : #11 ;
INFO  2017-02-11 06:19:52,000 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:52,000 [main][MJParser.java:588] : #8 }
INFO  2017-02-11 06:19:52,000 [main][MJParser.java:588] : #8 }
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog test9: notype, 0, 1 
   Meth someMethod: int, 0, 2 
      Var a: int, 0, 1 
      Var b: int, 1, 1 

   Meth someMethod2: notype, 5, 0 

   Meth main: notype, 11, 0 
      Var a: Arr of int, 0, 1 
      Var b: int, 1, 1 
      Var f: , 2, 1 



INFO  2017-02-11 06:19:52,001 [main][MJParser.java:636] : 
===============================
VARS COUNTING:
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:637] : Constants number: 0
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:638] : Global vars number: 0
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:639] : Global array vars number: 0
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:640] : Class vars number: 0
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:641] : Main function vars number: 3
===============================

INFO  2017-02-11 06:19:52,001 [main][MJParser.java:642] : 
===============================
FUNCTION CALLS COUNTING:
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:643] : Global functions: 3
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:644] : Class functions: 0
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:645] : Class static functions: 0
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:646] : Statements block number: 3
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:647] : Main method function calls number: 3
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:648] : Function formal parameters number: 2
INFO  2017-02-11 06:19:52,001 [main][MJParser.java:649] : Class definitions number: 0
